dimod/reference/composites/scalecomposite.py

```diff
@@ -0,0 +1,284 @@
+
+    def sample(self, bqm, scalar=None, bias_range=1, quadratic_range=None,
+               ignored_variables=None, ignored_interactions=None,
+               ignore_offset=False, **parameters):
+        """ Scale and sample from the provided binary quadratic model.
+
+        if scalar is not given, problem is scaled based on bias and quadratic
+        ranges. See :meth:`.BinaryQuadraticModel.scale` and
+        :meth:`.BinaryQuadraticModel.normalize`
+
+        Args:
+            bqm (:obj:`dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            scalar (number):
+                Value by which to scale the energy range of the binary quadratic model.
+
+            bias_range (number/pair):
+                Value/range by which to normalize the all the biases, or if
+                `quadratic_range` is provided, just the linear biases.
+
+            quadratic_range (number/pair):
+                Value/range by which to normalize the quadratic biases.
+
+            ignored_variables (iterable, optional):
+                Biases associated with these variables are not scaled.
+
+            ignored_interactions (iterable[tuple], optional):
+                As an iterable of 2-tuples. Biases associated with these interactions are not scaled.
+
+            ignore_offset (bool, default=False):
+                If True, the offset is not scaled.
+
+            **parameters:
+                Parameters for the sampling method, specified by the child sampler.
+
+        Returns:
+            :obj:`dimod.SampleSet`
+
+        """
+
+        child = self.child
+
+        bqm_copy = bqm.copy()
+        if scalar is None:
+            bqm_copy.normalize(bias_range, quadratic_range,
+                               ignored_variables=ignored_variables,
+                               ignored_interactions=ignored_interactions,
+                               ignore_offset=ignore_offset)
+        else:
+            bqm_copy.scale(scalar, ignored_variables=ignored_variables,
+                           ignored_interactions=ignored_interactions,
+                           ignore_offset=ignore_offset)
+        response = child.sample(bqm_copy, **parameters)
+
+        response.record.energy = bqm.energies((response.record.sample,
+                                               response.variables))
+        return response
+
+    def sample_ising(self, h, J, offset=0, scalar=None,
+                     bias_range=1, quadratic_range=None,
+                     ignored_variables=None, ignored_interactions=None,
+                     ignore_offset=False, **parameters):
+        """ Scale and sample from the problem provided by h, J, offset
+
+        if scalar is not given, problem is scaled based on bias and quadratic
+        ranges.
+
+        Args:
+            h (dict): linear biases corresponding to the HUBO form
+
+            J (dict): higher order biases corresponding to the HUBO form
```

Comment:
This should be less HUBO-specific. It might be worth just calling out at the end that this supports HUBOs